### DISCUSSION TITLE
Allow Autolink References with only numeric IDs

### DISCUSSION THREAD

**Discussion by micgro42** [community/community#29142] [Repositories] [answered]
Thank you for working on this "Autolink References" feature :pray:. However, the latest update, which makes all new [autolink configurations accept alphanumeric IDs](https://github.blog/changelog/2022-07-01-autolinks-with-alphanumeric-ids/), made this functionality basically useless to us.

We use Phabricator which uses plain IDs of the form `T123`, like https://phabricator.wikimedia.org/T1234. As you can see, the prefix is only `T`. This worked perfectly fine with the old configuration that allowed only numeric IDs, but with the new way of doing things, **all** words that start with a `T` or `t` become a reference and obviously almost all of them link nowhere.

Could you please give us the option to choose what characters are matched in the ID-part? Maybe being allowed to choose between "Numbers only (0-9)", "hexadecimal only (0-9a-f)" and "all alphanumeric characters (0-9a-z)", that would be great!

(Not sure if "Code Search and Navigation" is the best category for this, I found it fitting because this is about being able to navigate to related resources. Feel free to move it to a better category.)

**Comment by gm3dmo** [comment]
We've had a look at this behaviour and raised an internal issue for our engineering team to take a look. We'll provide an update when we have information to share.

**Reply by micgro42** [reply]
Thank you!

**Comment by vaindil** [comment, marked as answer]
Hey @micgro42, thanks for the report! Is the issue happening in repos that are part of a GitHub Enterprise Server instance, or is it happening on repos on github.com? I do see the problem on [wikimedia/ToolforgeBundle](https://github.com/wikimedia/ToolforgeBundle) ([example issue here](https://github.com/wikimedia/ToolforgeBundle/issues/25)), and it looks like the autolink was added after this change (since it wasn't opted into the legacy behavior of only matching numeric values). Still researching the issue though.

**Reply by micgro42** [reply]
Thank you for looking into this, @vaindil 🙏.

We're only using github.com, not the GitHub Enterprise Server.

For an example where we're using the legacy behavior, I would point you to [wmde/new-lexeme-special-page](https://github.com/wmde/new-lexeme-special-page) ([example issue](https://github.com/wmde/new-lexeme-special-page/pull/254)).

**Reply by vaindil** [reply]
Gotcha, thanks! We're currently working on adding a checkbox to allow people to control whether autolinks are numeric or alphanumeric based on this and some other feedback, I'll report back when that's live.

**Reply by micgro42** [reply]
Thank you! I'll look forward to your update :)

**Reply by vaindil** [reply]
This just went live, you can now choose whether an autolink is alphanumeric or numeric when you create it. Thanks again for the feedback!

**Reply by micgro42** [reply]
It works! Thank you so much!

**Comment by BYK** [comment]
I'd also like to add another case where we are now having issues with our Jira and Sentry auto links:

Jira uses numeric ids, Sentry does not. We added auto linking for Jira whose prefix collides with our Sentry project prefixes. We were relying on a [custom action to autolink to Sentry](https://github.com/formsort/sentryfuge/) but now our Sentry references get linked to Jira incorrectly. We'd love the option to have "numeric ids only".

**Reply by vaindil** [reply]
We just shipped this feature, you can now choose whether an autolink is alphanumeric or numeric when you create it. Thank you for the feedback!
